x/transferinject/ibc_middleware_test.go

```diff
@@ -0,0 +1,461 @@
+package transferinject_test
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+	"github.com/stretchr/testify/require"
+
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject/types"
+)
+
+func TestIBCMiddleware_SendPacket(t *testing.T) {
+	type fields struct {
+		ICS4Wrapper   porttypes.ICS4Wrapper
+		rollappKeeper types.RollappKeeper
+		bankKeeper    types.BankKeeper
+	}
+	tests := []struct {
+		name         string
+		fields       fields
+		data         []byte
+		wantSentData []byte
+		wantErr      string
+	}{
+		{
+			name: "success: added denom metadata to memo",
+			fields: fields{
+				ICS4Wrapper: &mockICS4Wrapper{},
+				rollappKeeper: &mockRollappKeeper{
+					returnRollapp: &rollapptypes.Rollapp{},
+				},
+				bankKeeper: mockBankKeeper{
+					returnMetadata: validDenomMetadata,
+				},
+			},
+			data: types.ModuleCdc.MustMarshalJSON(&transfertypes.FungibleTokenPacketData{
+				Denom: "adym",
```

Comment:
why not put the types.ModuleCdc.MustMarshalJSON calls once, in the execution part, and make the data field the &transfertypes.FungibleTokenPacketDat type?
Less characters oeverall